modules/light-clients/07-tendermint/light_client_module.go

```diff
@@ -41,7 +43,7 @@ func (l *LightClientModule) RegisterStoreProvider(storeProvider exported.ClientS
 func (l LightClientModule) Initialize(ctx sdk.Context, clientID string, clientStateBz, consensusStateBz []byte) error {
 	var clientState ClientState
 	if err := l.keeper.Codec().Unmarshal(clientStateBz, &clientState); err != nil {
-		return err
+		return fmt.Errorf("failed to unmarshal client state bytes into client state: %w", err)
```

Comment:
I think I might change this in a follow-up PR, because at the moment we are not consistent with the error we return in these situations for other light clients (in some places we do `return err` but in other places we do `errorsmod.Wrap(clienttypes.ErrInvalidClient, err.Error())`...

Reply:
we mostly return this kind of nonspecific error for unmarshalling/proof verification right now. i guess consistency makes sense but i think for testing/debugging purposes this error also works!